pkg/volume/csi/csi_mounter.go

```diff
@@ -277,17 +278,22 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 		klog.V(2).Info(log("error checking for SELinux support: %s", err))
 	}
 
-	// apply volume ownership
-	// The following logic is derived from https://github.com/kubernetes/kubernetes/issues/66323
-	// if fstype is "", then skip fsgroup (could be indication of non-block filesystem)
-	// if fstype is provided and pv.AccessMode == ReadWriteOnly, then apply fsgroup
-	err = c.applyFSGroup(fsType, mounterArgs.FsGroup, mounterArgs.FSGroupChangePolicy)
-	if err != nil {
-		// At this point mount operation is successful:
-		//   1. Since volume can not be used by the pod because of invalid permissions, we must return error
-		//   2. Since mount is successful, we must record volume as mounted in uncertain state, so it can be
-		//      cleaned up.
-		return volumetypes.NewUncertainProgressError(fmt.Sprintf("applyFSGroup failed for vol %s: %v", c.volumeID, err))
+	// check to determine if the feature gate is disabled (proceed to default behavior)
+	// if the feature gate is enabled, then ensure that the SupportsFSGroup != Never
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIVolumeSupportsFSGroup) ||
+		c.SupportsFSGroup != storage.NeverSupportsFSGroup {
+
+		// The following logic is derived from https://github.com/kubernetes/kubernetes/issues/66323
```

Comment:
Should this comment go inside `applyFSGroup`?

Reply:
The comment is currently included in both places - I don't mind removing it, but to me it makes sense to include it in both places, as it can easily be referenced regardless of location without adding too much bloat.